.github/workflows/build_push_img.yml

```diff
@@ -0,0 +1,135 @@
+name: Build and Push Marqo Docker Image
+
+on:
+  workflow_dispatch:
+    inputs:
+      marqo_ref:
+        description: 'Marqo branch-name, commit SHA or tag'
+        required: false
+        default: 'mainline'
+      push_to:
+        description: 'image destination. Options: "ECR", "DockerHub"'
+        required: true
+        options:
+          - ECR
+          - DockerHub
+      image_repo:
+        description: 'Image repository'
+        required: true
+        default: 'marqo'
+      image_tag:
+        description: 'Image tag'
+        required: true
+      dockerhub_username:
+        description: 'DockerHub username'
+        required: false
+      dockerhub_password:
+        description: 'DockerHub password'
+        required: false
+
+jobs:
+  Start-Runner:
+    # on US-WEST-2, open source account
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.MARQO_WORKFLOW_TESTS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.MARQO_WORKFLOW_TESTS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          # 200 GB amd64 image
+          ec2-image-id:  ami-01accacd51bdde263
+          ec2-instance-type: t3.xlarge
+          subnet-id: subnet-038b1447ac3c97e7a
+          security-group-id: sg-094be521399e0d5ba
```

Comment:
Maybe add some tags like this? https://github.com/marqo-ai/marqo/blob/5c372ddc48498a50ea1fb8f53364644f6fbefcaa/.github/workflows/unit_test_200gb_CI.yml#L47-L55

Reply:
Good idea, done.